### PR TITLE
[cors] Allow any valid origin type in dynamic origin callback

### PR DESCRIPTION
**Pull request by CleyFaye**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/expressjs/cors#configuring-cors-w-dynamic-origin
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by pluma** [comment]
@gtpan77 Can you review this? I'm not using this library at the moment and don't have the time.

**Comment by gtpan77** [comment]
Sure @pluma, I'll review this

File: types/cors/index.d.ts [hunk 1]
```diff
@@ -7,7 +7,9 @@
 
 import { IncomingHttpHeaders } from 'http';
 
-type CustomOrigin = (requestOrigin: string | undefined, callback: (err: Error | null, allow?: boolean) => void) => void;
+type StaticOrigin = boolean | string | RegExp | (string | RegExp)[];
+
+type CustomOrigin = (requestOrigin: string | undefined, callback: (err: Error | null, origin?: StaticOrigin) => void) => void;
```
Comment (gtpan77) [inline types/cors/index.d.ts:12]: `allow?: boolean` is getting changed to `origin?: StaticOrigin`.

**Review by gtpan77 — APPROVED** [review]
LGTM

**Comment by gtpan77** [comment]
Ready to merge

**Pull request merged**